src/ringmesh/basic/attributes.cpp

```diff
@@ -0,0 +1,550 @@
+    {
+    public:
+        ReadOnlyScalarAttributeAdapterET_UINT8(
+            const AttributesManager& manager, const std::string& name )
+            : ReadOnlyScalarAttributeAdapter( manager, name )
+        {
+        }
+
+        double operator[]( index_t i ) override
+        {
+            return get_element< uint8 >( i );
+        }
+
+        index_t nb_scalar_elements_per_item() const override
+        {
+            return 1;
+        }
+        /* AttributeElementType element_type() const override
+         {
+             return typeid( uint8 ).name();
+         }*/
+        bool is_integer_like_attribute() const override
+        {
+            return true;
+        }
+    };
+
+    class ReadOnlyScalarAttributeAdapterET_INT8
+        : public ReadOnlyScalarAttributeAdapter
+    {
+    public:
+        ReadOnlyScalarAttributeAdapterET_INT8(
+            const AttributesManager& manager, const std::string& name )
+            : ReadOnlyScalarAttributeAdapter( manager, name )
+        {
+        }
+
+        double operator[]( index_t i ) override
+        {
+            return get_element< int8 >( i );
+        }
+
+        index_t nb_scalar_elements_per_item() const override
+        {
+            return 1;
+        }
+        /*AttributeElementType element_type() const override
+        {
+            return typeid( int8 ).name();
+        }*/
+        bool is_integer_like_attribute() const override
+        {
+            return true;
+        }
+    };
+
+    class ReadOnlyScalarAttributeAdapterET_UINT32
+        : public ReadOnlyScalarAttributeAdapter
+    {
+    public:
+        ReadOnlyScalarAttributeAdapterET_UINT32(
+            const AttributesManager& manager, const std::string& name )
+            : ReadOnlyScalarAttributeAdapter( manager, name )
+        {
+        }
+
+        double operator[]( index_t i ) override
+        {
+            return get_element< uint32 >( i );
+        }
+
+        index_t nb_scalar_elements_per_item() const override
+        {
+            return 1;
+        }
+        /*AttributeElementType element_type() const override
+        {
+            return typeid( unsigned int ).name();
+        }*/
+        bool is_integer_like_attribute() const override
+        {
+            return true;
+        }
+    };
+
+    class ReadOnlyScalarAttributeAdapterET_INT32
+        : public ReadOnlyScalarAttributeAdapter
+    {
+    public:
+        ReadOnlyScalarAttributeAdapterET_INT32(
+            const AttributesManager& manager, const std::string& name )
+            : ReadOnlyScalarAttributeAdapter( manager, name )
+        {
+        }
+
+        double operator[]( index_t i ) override
+        {
+            return get_element< int32 >( i );
+        }
+
+        index_t nb_scalar_elements_per_item() const override
+        {
+            return 1;
+        }
+        /*AttributeElementType element_type() const override
+        {
+            return typeid( int ).name();

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this block of commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3ACommentedCode)